search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -95,20 +94,27 @@ class TemporalEntityHandler(
                 .body(BadRequestDataResponse(e.message))
         }
 
-        val temporalEntityAttribute = temporalEntityAttributeService.getForEntity(
+        val temporalEntityAttributes = temporalEntityAttributeService.getForEntity(
             entityId.toUri(),
             temporalQuery.attrs,
             contextLink
-        ).awaitFirstOrNull() ?: throw ResourceNotFoundException(entityNotFoundMessage(entityId))
+        ).collectList().awaitFirst()
 
-        val results = attributeInstanceService.search(temporalQuery, temporalEntityAttribute).awaitFirst()
+        if (temporalEntityAttributes.isEmpty()) throw ResourceNotFoundException(entityNotFoundMessage(entityId))
 
-        val jsonLdEntity = loadEntityPayload(temporalEntityAttribute, bearerToken).awaitFirst()
+        val attributeAndResultsMap = temporalEntityAttributes.map { attribute ->
```

Comment:
temporalEntityAttribute ->

Reply:
I thought it was too long name, I'm going to use "it"  if you don't mind